File: src/SampleCRM/Views/BaseUserControl.cs
using System;
using System.ComponentModel;
using System.Runtime.CompilerServices;
using System.Windows;
using System.Windows.Controls;

namespace SampleCRM.Web.Views
{
    public class BaseUserControl : UserControl
    {
        protected const string _imageFileExtFilter = "All Images Files (*.png;*.jpeg;*.gif;*.jpg;*.bmp;*.tiff;*.tif)|*.png;*.jpeg;*.gif;*.jpg;*.bmp;*.tiff;*.tif" +
            "|PNG Portable Network Graphics (*.png)|*.png" +
            "|JPEG File Interchange Format (*.jpg *.jpeg *jfif)|*.jpg;*.jpeg;*.jfif" +
            "|BMP Windows Bitmap (*.bmp)|*.bmp" +
            "|TIF Tagged Imaged File Format (*.tif *.tiff)|*.tif;*.tiff" +
            "|GIF Graphics Interchange Format (*.gif)|*.gif";
        
        protected virtual double MaxMobileWidth => 700d;

        public BaseUserControl()
        {
            Loaded += BaseUserControl_Loaded;
            Application.Current.MainWindow.SizeChanged += MainWindow_SizeChanged;
        }

        private void MainWindow_SizeChanged(object sender, WindowSizeChangedEventArgs e)
        {
            IsMobileUI = Application.Current.MainWindow.ActualWidth <= MaxMobileWidth;
            ArrangeLayout();
        }

        private void BaseUserControl_Loaded(object sender, RoutedEventArgs e)
        {
            ArrangeLayout();
        }

        public virtual void ArrangeLayout() { }

        public bool IsMobileUI
        {
            get { return (bool)GetValue(IsMobileUIProperty); }
            set { SetValue(IsMobileUIProperty, value); }
        }
        public static readonly DependencyProperty IsMobileUIProperty =
            DependencyProperty.Register("IsMobileUI", typeof(bool), typeof(BaseUserControl), new PropertyMetadata(null));
    }
}

File: src/SampleCRM/Views/BaseUserControl.vb
Imports System.ComponentModel
Imports System.Runtime.CompilerServices
Imports System.Windows
Imports System.Windows.Controls

Namespace Views
    Public Class BaseUserControl
        Inherits UserControl
        Implements INotifyPropertyChanged

        Protected Overridable ReadOnly Property MaxMobileWidth As Double
            Get
                Return 700.0R
            End Get
        End Property

        Public Sub New()
            AddHandler Loaded, AddressOf BaseUserControl_Loaded
            AddHandler Application.Current.MainWindow.SizeChanged, AddressOf MainWindow_SizeChanged
        End Sub

        Private Sub MainWindow_SizeChanged(sender As Object, e As WindowSizeChangedEventArgs)
            ArrangeLayout()
            OnPropertyChanged(NameOf(IsMobileUI))
        End Sub

        Private Sub BaseUserControl_Loaded(sender As Object, e As RoutedEventArgs)
            ArrangeLayout()
        End Sub

        Public Overridable Sub ArrangeLayout()
        End Sub

        Public ReadOnly Property IsMobileUI As Boolean
            Get
                Return Application.Current.MainWindow.ActualWidth <= MaxMobileWidth
            End Get
        End Property

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Protected Sub OnPropertyChanged(<CallerMemberName> Optional propertyName As String = "")
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub
    End Class
End Namespace